### PR TITLE
docs: update vagrant setup

### PR DESCRIPTION
**Pull request by devadvocado**
### Full changelog

* [Implement ...]
* [Fix ...]

### Issues resolved

Fix #XXX

### CODE REVIEW

File: vagrant/README.md [hunk 1]
```diff
@@ -116,26 +171,109 @@ default   backend    service=backend version=v0   Online   6m3s
...
-To shop at Kuma's marketplace, access the Kong gateway that is the ingress to your mesh at [http://192.168.33.70:8000](http://192.168.33.70:8000). All the traffic between the machines are routed through Kuma's dataplane.
+Use `kumactl apply [..]` to enable mTLS on our mesh.
 
-### 10. Let's enable mTLS using `kumactl`:
-
-```
+```bash
 $ cat <<EOF | kumactl apply -f -
-type: Mesh
+type: mesh
```
Comment (yskopets) [inline vagrant/README.md]: Let's revert this change:
* `type: mesh` is NOT valid because `Mesh` must begin with a capital `M`

File: vagrant/README.md [hunk 2]
```diff
@@ -144,18 +282,27 @@ mtls:
...
-If you try to access the marketplace via [http://192.168.33.70:8000](http://192.168.33.70:8000), it won't work because that traffic goes through the dataplane and is now encrypted via mTLS.
+If you try to access the marketplace via [http://192.168.33.70:8000](http://192.168.33.70:8000), it won't work because that traffic goes through the dataplane and is now encrypted via mTLS. 
 
-### 11. Now let's enable traffic-permission for all services so our application will work like it use to:
-```
+To enable traffic once mTLS has been enabled, please add [traffic permission policies](#traffic-permissions-policy).
+
+### Traffic Permissions
+
+Traffic Permissions allow you to determine security rules for services that consume other services via their Tags. It is a very useful policy to increase security in the mesh and compliance in the organization. You can determine what source services are allowed to consume specific destination services. The service field is mandatory in both sources and destinations.
```
Comment (yskopets) [inline vagrant/README.md]: > Traffic Permissions allow you to determine security rules for services that consume other services via their Tags.

Too complicated

File: vagrant/README.md [hunk 3]
```diff
@@ -291,11 +438,49 @@ EOF
...
+A user can enable traffic metrics by editing a mesh resource and providing the desired mesh-wide configuration. If necessary, metrics configuration can be customized for each Dataplane individually, e.g. to override the default metrics port that might be already in use on that particular machine. 
+
+#### Adding Traffic Metric Policy
+
+Let's enable traffic metrics by editing our mesh resource like so:
+
+```bash
 $ cat <<EOF | kumactl apply -f -
-type: Mesh
+type: mesh
```
Comment (yskopets) [inline vagrant/README.md]: Let's revert this change:
* `type: mesh` is NOT valid because `Mesh` must begin with a capital `M`

File: vagrant/README.md [hunk 4]
```diff
@@ -1,42 +1,86 @@
...
 
 1. The first machine hosts the Kuma control plane.
-2. The second machine will host our frontend application that allows you to visually interact with the marketplace
-3. The third machine will host our backend application that handles the logic of our application
-4. The fourth machine will host the Elasticsearch service that stores all the items in our marketplace
-5. The fifth machine will host Redis service that stores reviews for each item
-6. The sixth machine will be the Kong Gateway acting as an ingress to your mesh
+2. The second machine will host Redis service that stores reviews for each item
+3. The third machine will host the Elasticsearch service that stores all the items in our marketplace
+4. The fourth machine will host the Prometheus dashboard and the [kuma-prometheus-sd](https://kuma.io/docs/0.3.2/policies/#traffic-metrics)
```
Comment (yskopets) [inline vagrant/README.md]: Let's refer to `latest` version instead of `0.3.2`

File: vagrant/README.md [hunk 5]
```diff
@@ -116,26 +171,109 @@ default   backend    service=backend version=v0   Online   6m3s
...
+### GUI
+
+Kuma ships with an internal GUI that will help you visualize the mesh and its policies in an intuitive format. It can be found on port `:5683` on the control-plane machine. Since our Kuma control-plane machine's IP is `192.168.33.10`, navigate to [http://192.168.33.10:5683/](http://192.168.33.10:5683/) to use Kuma's GUI.
+
+
+## Integrations
+
+### Kong Gateway
+
+In this Vagrant demo, Kong is already configured and deployed when you ran `vagrant up` in the [vagrant setup step](#vagrant). But to give you an in depth dive on how it works, lets dig into the files in the [`/vagrant/kong/`](/vagrant/kong/) directory.
```
Comment (yskopets) [inline vagrant/README.md]: Why all `/vagrant/...` paths are absolute ? It's confusing.

Let's change `[/vagrant/kong/](/vagrant/kong/)` to `[kong/](./kong/)`

File: vagrant/README.md [hunk 6]
```diff
@@ -116,26 +171,109 @@ default   backend    service=backend version=v0   Online   6m3s
...
+  routes:
+  - name: frontend-root-route
+    paths:
+    - /
+```
+We start with a Service; that is the name Kong uses to refer to the upstream APIs and microservices it manages. Marketplace's [frontend service](/README.md#frontend) is the point of entry to our application so we add that as our service. Before you can start making requests against this service, you will need to add a route to it. Routes specify how (and if) requests are sent to their services after they reach Kong. A single Service can have many Routes. We'll add a route called `frontend-root-route` that points any request to the `/` endpoint to our frontend service.
+
+#### Configure Dataplane with Gateway mode
+
+When you use a dataplane with a service, both inbound traffic to a service and outbound traffic from the service flows through the dataplane. But with Kong deployed, we want inbound traffic to go directly to API Gateway, otherwise clients would have to be provided with certificates that are generated dynamically for communication between services within the mesh. So we have to operate the dataplane in [Gateway mode](https://kuma.io/docs/0.3.2/documentation/#gateway).
```
Comment (yskopets) [inline vagrant/README.md]: Let's use `latest` version instead of `0.3.2`

File: vagrant/README.md [hunk 7]
```diff
@@ -116,26 +171,109 @@ default   backend    service=backend version=v0   Online   6m3s
...
+
+With this dataplane policy, [`register_dataplane.sh`](/vagrant/common/register_dataplane.sh) will apply this configuration to the dataplane that will sit alongside our Kong gateway. 
+
+### Prometheus
+
+Out-of-the-box, Kuma provides full integration with Prometheus:
+* if enabled, every dataplane will expose its metrics in Prometheus format
+* furthemore, Kuma will make sure that Prometheus can automatically find every dataplane in the mesh
+
+In this demo guide, Prometheus is already deployed and configured. All the files that are used to configure and deploy Prometheus can be found in the [`/vagrant/prometheus/`](/vagrant/prometheus/app/install.sh) directory. If you need more help, refer to the official documentation [here](https://kuma.io/docs/0.3.2/policies/#traffic-metrics). **To enable Prometheus metrics on every dataplane in the mesh, configure a mesh resource as shown in the [Traffic Metric Policy](#traffic-metrics) section.**
```
Comment (yskopets) [inline vagrant/README.md]: Let's use `latest` version instead of `0.3.2`

File: vagrant/README.md (hunk 1 above)
Comment (devadvocado) [reply vagrant/README.md]: ahh i did a mass replace so these got changed, fixed

File: vagrant/README.md (hunk 4 above)
Comment (yskopets) [reply vagrant/README.md]: I still see `0.3.2` instead of `latest`
